src/account-info/CustomProfileFields.js

```diff
@@ -0,0 +1,129 @@
+// @flow strict-local
+import * as React from 'react';
+import { View } from 'react-native';
+
+import { type UserOrBot, type UserId } from '../api/modelTypes';
+import WebLink from '../common/WebLink';
+import ZulipText from '../common/ZulipText';
+import ZulipTextIntl from '../common/ZulipTextIntl';
+import { ensureUnreachable } from '../generics';
+import { useSelector } from '../react-redux';
+import { tryGetUserForId } from '../selectors';
+import {
+  type CustomProfileFieldValue,
+  getCustomProfileFieldsForUser,
+} from '../users/userSelectors';
+import UserItem from '../users/UserItem';
+import * as NavigationService from '../nav/NavigationService';
+import { navigateToAccountDetails } from '../nav/navActions';
+
+/* eslint-disable no-shadow */
+
+type Props = {|
+  +user: UserOrBot,
+|};
+
+function CustomProfileFieldUser(props: {| +userId: UserId |}): React.Node {
+  const { userId } = props;
+  const user = useSelector(state => tryGetUserForId(state, userId));
+
+  const onPress = React.useCallback((user: UserOrBot) => {
+    NavigationService.dispatch(navigateToAccountDetails(user.user_id));
```

Comment:
Where reasonable, it'd be great to avoid bringing `NavigationService` into more files, for #4417. Unfortunately, both before and after #5397, I've had trouble with Flow not accepting a `useNavigation()` with `navigation.push('account-details', { userId })`, even though I think it should be fine. Do you see an easy way to do that?

After #5397, a `useNavigation()` with `navigation.dispatch(navigateToAccountDetails(user.user_id))` seems to work. But that doesn't help us shrink `navActions` for #4417.

Reply:
> Unfortunately, both before and after #5397, I've had trouble with Flow not accepting a `useNavigation()` with `navigation.push('account-details', { userId })`, even though I think it should be fine. Do you see an easy way to do that?

Hm. Trying it, Flow gives the error that property `push` is missing on the type of `navigation`.

I think basically the issue is that `useNavigation` returns a plain `NavigationProp`, but we want specifically a `StackNavigationProp`. That should be fixable in our type-wrapper.

---

Though the other thing about actually cutting out `navActions` is that we don't have useful type-checking on those `StackActions.push` calls, of the fact that the parameters they pass align with what the screens actually expect. Keeping them all centralized in one place is a way of mitigating that, by making it easier to more reliably check them by hand. See: https://github.com/zulip/zulip-mobile/pull/4757#discussion_r637180840 (which has been on my to-do list to write up.)

Hmm, but I guess that problem might be solvable precisely by saying `navigation.push` instead of `StackActions.push`! Because the `navigation` object should have a type tying it to our `GlobalParamsList` which registers all those route-params types. I'll look into that.

Even excluding the latter point: if the goal is to eliminate dispatching of nav actions and instead use `navigation.push`, and if we still needed a central set of functions to mitigate a lack of type-checking, we could still accomplish that. It's just that the central set of functions wouldn't be returning actions, and instead would have signatures like `navigateToAccountDetails: (navigation: NavigationProp<…>, userId: UserId) => void` and would make the `navigation.push` calls themselves.

---

Anyway: I think for this PR, I'll switch to `useNavigation` but leave it as a `navigation.dispatch`. Then I'll look into making `useNavigation` return a `StackNavigationProp`, and converting this and others to say `navigation.push` directly.

Reply:
(That followup may be helpful toward the react-navigation v6 upgrade #4936, too.)